### ISSUE TITLE
Error 520 on WP site using Cloudflare CDN

### ISSUE THREAD

**Issue by gasmas75** [engintron/engintron#1229]
Hello,
Possibly not an Engintron issue, since it's running for two years now without problems. But 5 days ago 520 error began to show up on wordpress sites that are hosted and distributed via CF. Their support told me that their logs showed these records:
```
message:upstream prematurely closed connection while reading response header from upstream
upstream:https://xxx.xx.xx.xxx:443/wp-admin/admin-ajax.php
```
Now my logs (nginx, apache, sql) don't indicate anything wrong near or on these times that the error occurs. In fact Nginx log is crystal clear.

A research on the error shows that this is a Nginx error and some suggestions say to increase the proxy timeout settings
```
    proxy_connect_timeout 1000;
    proxy_send_timeout 1000;
    proxy_read_timeout 1000;
    send_timeout 1000;
```

Do you this that this would help? And why didn't this occur earlier if this a Nginx issue?
If putting these directives, do help, should I add them on the Custom Rules section of the WHM Engintron Plugin?

Some system info:
CentosL 7.8
WHM: v90.0.8
PHP: 7.4
EasyApache4 with Engintron_EA4_2019_v1 profile modified to include PHP 7.4
Engintron: 1.13.0

Any help is appreciated
Thank you.
Kostas

**Comment by fevangelou** [comment]
Nginx already has timeout values of 300 sec (in line with Apache's timeout value). That's more than enough for anything.

You should look into optimizing your server's components: Apache, MySQL/MariaDB, using some cache driver like APCu or Memcached etc. etc.

If you ask why the sudden change, well, if you're on a VPS, it's most likely that your host server now has more VPSs to run or you got busy neighbours. You can double check that by installing Munin. If you don't see any CPU loads on Munin after a week but your sites are slow or timeout, it could be a "noisy neighbour" issue...

**Comment by gasmas75** [comment]
Hi Fotis,
Thank you for your reply. Sorry I didn't clarify those aspects on my initial post. It's not a VPS, it's a dedicated server (or at least that's what the hosting claims). Munin is installed from the very beginning, following Engintron wiki suggestion and the monitoring stats are as normal as hell.
But...
After researching I found and resolved the issue.
I had to configure the httpd.conf for remote_ip module to restore the original visitor IPs.
I added the directives on WHM Home->Service Configuration->Apache Configuration->Include Editor --> Pre Main Include
The directives are located here : [https://support.cloudflare.com/hc/en-us/articles/360029696071-Restoring-original-visitor-IPs-Option-2-Installing-mod-remoteip-with-Apache](url)

Hope this helps someone
Thank you
Kostas
